test/ScopeExitTest.cpp

```diff
@@ -0,0 +1,44 @@
+#include "pch.h"
+#include <kf/ScopeExit.h>
+
+SCENARIO("SCOPE_EXIT macro")
+{
+    GIVEN("A block of code with SCOPE_EXIT")
+    {
+        WHEN("The block is executed")
```

Comment:
Let's also check that the function is not called right away. We had this test in the scenario without the macro.